typescript-generator-gradle-plugin/src/main/java/cz/habarta/typescript/generator/gradle/TypeScriptGeneratorPlugin.java

```diff
@@ -16,6 +16,7 @@ public void apply(Project project) {
         for (Task task : project.getTasks()) {
```

Comment:
To facilitate configuration avoidance and lazy task initialization:
```
project.getTasks().matching{ it.getName().startsWith('compile') }.all { Task task ->
    generateTsTask.getInputs().files(task)
}
```
Adding the task directly as input implicitly creates a task dependency and an explicit dependsOn is not needed.
The `matching.all` construct does not materialize the tasks and task initialization is deferred.

Reply:
@rbrindl It's still require to upgrade gradle due different signatures: https://docs.gradle.org/2.1/javadoc/org/gradle/api/tasks/TaskInputs.html#files(java.lang.Object...) and
https://docs.gradle.org/3.0/javadoc/org/gradle/api/tasks/TaskInputs.html#files(java.lang.Object...)
Without it I get error:
`Caused by: java.lang.NoSuchMethodError: org.gradle.api.tasks.TaskInputs.files([Ljava/lang/Object;)Lorg/gradle/api/tasks/TaskInputs;`